src/cryptsetup/cryptsetup.c

```diff
@@ -38,7 +38,15 @@
 #include "strv.h"
 #include "util.h"
 
-static const char *arg_type = NULL; /* CRYPT_LUKS1, CRYPT_TCRYPT or CRYPT_PLAIN */
+/* libcryptsetup define for any LUKS version, compatible with libcryptsetup 1.x */
+#ifndef CRYPT_LUKS
+#define CRYPT_LUKS NULL
+#endif
+
+/* internal helper */
+#define ANY_LUKS "LUKS"
```

Comment:
hmm, if "LUKS" is understood by libcryptsetup, why isn't this included in the header files as a `#define` of its own?

Reply:
No, it's not defined in libcryptsetup. I added the helper just because arg_type == NULL is interpreted differently by systemd-cryptsetup utility. IIUC it tries to load CRYPT_PLAIN if previous crypt_load(cd, CRYPT_LUKS1, NULL) (or any version with my patch) attempt failed.